src/main/kotlin/com/ebay/plugins/metrics/develocity/TaskProviderExtensions.kt

```diff
@@ -39,39 +29,43 @@ fun TaskProvider<out MetricSummarizerTask>.inputsFromDuration(
     project: Project,
     durationSpec: String,
     summarizerId: String,
-) {
-    val inputTask = project.internalExt().createDurationTask(durationSpec)
-    configureInputs(inputTask, summarizerId)
-}
-
-/**
- * Get the internal extension for a given project.
- */
-private fun Project.internalExt(): MetricsForDevelocityInternalExtension {
-    /*
-     * NOTE: This is a bit of a project isolation violation.  We could likely avoid this if task
-     * rules applied to programmatically created tasks as well as those created on the command
-     * line.  This works for now...
-     */
-    // TODO: This violates project isolation
-    val ext = project.rootProject.extensions.getByType(MetricsForDevelocityExtension::class.java)
-    return ext.extensions.getByType(MetricsForDevelocityInternalExtension::class.java)
-}
+) = configureInputs(project, durationSpec, summarizerId)
 
 /**
  * Common helper configuration for tasks which consume the summarizer output.
  */
 private fun TaskProvider<out MetricSummarizerTask>.configureInputs(
-    inputTask: TaskProvider<out MetricsIntermediateTask>,
+    project: Project,
+    timeSpec: String,
     summarizerId: String,
 ) {
+    val resolveId = "$name-resolve-$summarizerId"
+    val existingConfig = project.configurations.findByName(resolveId)
+    val configProvider: Provider<Configuration> = if (existingConfig == null) {
+        val resolveConfig = project.configurations.register(resolveId)
+        resolveConfig.configure { config ->
+            with(config) {
+                isTransitive = false
+                isCanBeResolved = true
+                isCanBeConsumed = false
+                attributes.attribute(TIME_SPEC_ATTRIBUTE, timeSpec)
+                attributes.attribute(SUMMARIZER_ATTRIBUTE, SUMMARIZER_ALL)
+                dependencies.add(project.dependencies.project(mapOf("path" to ":")))
+            }
+        }
+        resolveConfig
+    } else {
+        project.provider { existingConfig }
+    }
+
     configure { self ->
         with(self) {
-            dependsOn(inputTask)
-            val inputFileProvider = inputTask.flatMap { aggregateTask ->
-                PathUtil.summarizerFile(aggregateTask.outputDirectoryProperty, summarizerId)
+            dependsOn(configProvider)
+            val summaryFileProvider = configProvider.get().incoming.files.elements.map { files ->
+                files.map { it.asFile }.firstOrNull()?.resolve(summarizerId)
+                    ?: throw GradleException("Could not find summarizer output file '${summarizerId}'")
```

Comment:
This deviates from the previously attempted approach.

Previously, we used `artifactView` and registered/supplied an artifact transform to select the individual summarizer result file from the base artifact.  I cannot get this approach to work from an API perspective and so I resorted to inlining the logic in order to avoid the transform.

I need to create a repro and raise this issue with Gradle...